Userland/Libraries/LibC/unistd.cpp

```diff
@@ -398,11 +401,71 @@ ssize_t pwrite(int fd, const void* buf, size_t count, off_t offset)
     return nwritten;
 }
 
+// Note: Be sure to send to directory_name parameter a directory name ended with trailing slash.
+static int ttyname_r_for_directory(const char* directory_name, dev_t device_mode, ino_t inode_number, char* buffer, size_t size)
+{
+    DIR* dirstream = opendir(directory_name);
+    if (!dirstream) {
+        return -1;
+    }
+    struct stat st;
+    struct dirent* entry;
+    char* name_path;
+
+    while ((entry = readdir(dirstream)) != nullptr) {
```

Comment:
Not that we should care about it, but I think using `readdir` here breaks the reentrancy guarantees of `ttyname_r`.

Reply:
So after you explained we link LibC against libstdc++ and therefore we can use C++ code, this seems like a good idea, but probably not suitable for this PR. A FIXME will be added though to keep note of this.